Unable to resolve dependency tree when running `npm run test:ghpages:beta`

<!--
  - Thanks for opening an issue for us!  Before you open an issue,
  - please check if a similar issue exists or has been closed before.
  -
  - If you're asking a question about how to use Blockly in your application,
  - please ask questions on the mailing list, instead of filing issues:
  -       https://groups.google.com/forum/#!forum/blockly
  -->

**Category**
Plugins
<!-- Choose one of:
 - Plugins
 - Examples
 - Codelabs
-->

**Component**
All plugins
<!-- The name of the component that has a bug (e.g. field-date, blockly-rtc).-->

**Describe the bug**
When the command `npm run test:ghpages:beta` is run, it fails on the step `npm exec --npm install blockly@beta` with an error about being unable to resolve the dependency tree.

<details>
  <summary>Expand for stack trace</summary>
  
```
[kozbial@kozbial samples-moniika](2021_q4_release)$ npm run test:ghpages:beta 

> blockly-samples@0.0.0 test:ghpages:beta
> gulp testGhPagesBeta

[16:22:49] Using gulpfile ~/github/blockly/samples-moniika/gulpfile.js
[16:22:49] Starting 'testGhPagesBeta'...
[16:22:49] Starting '<anonymous>'...
[16:22:49] Starting '<anonymous>'...
lerna notice cli v3.20.2
lerna info versioning independent
lerna info Executing command in 32 packages: "npm install blockly@beta"

up to date, audited 112 packages in 560ms

2 packages are looking for funding
  run `npm fund` for details
⸨⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⸩ ⠼ idealTree:node: sill idealTree buildDeps
found 0 vulnerabilities
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: @blockly/plugin-workspace-search-example-node@undefined
npm ERR! Found: blockly@7.20211209.0-beta.0
npm ERR! node_modules/blockly
npm ERR!   blockly@"7.20211209.0-beta.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer blockly@">=3.20200123.0 <4" from @blockly/plugin-workspace-search@1.20200312.2
npm ERR! node_modules/@blockly/plugin-workspace-search
npm ERR!   @blockly/plugin-workspace-search@"1.20200312.2" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /usr/local/google/home/kozbial/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /usr/local/google/home/kozbial/.npm/_logs/2021-10-01T23_22_51_410Z-debug.log
lerna ERR! npm install blockly@beta exited 1 in '@blockly/plugin-workspace-search-example-node'
lerna ERR! npm install blockly@beta exited 1 in '@blockly/plugin-workspace-search-example-node'
lerna WARN complete Waiting for 3 child processes to exit. CTRL-C to exit immediately.
[16:22:51] '<anonymous>' errored after 1.79 s
[16:22:51] Error: Command failed: lerna exec -- npm install blockly@beta
    at checkExecSyncError (node:child_process:826:11)
    at execSync (node:child_process:900:15)
    at /usr/local/google/home/kozbial/github/blockly/samples-moniika/gulpfile.js:281:7
    at bound (node:domain:421:15)
    at runBound (node:domain:432:12)
    at asyncRunner (/usr/local/google/home/kozbial/github/blockly/samples-moniika/node_modules/async-done/index.js:55:18)
    at processTicksAndRejections (node:internal/process/task_queues:78:11)
[16:22:51] 'testGhPagesBeta' errored after 1.79 s
[16:22:51] The following tasks did not complete: <anonymous>
[16:22:51] Did you forget to signal async completion?
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: @blockly/plugin-workspace-search-example-script@undefined
npm ERR! Found: blockly@7.20211209.0-beta.0
npm ERR! node_modules/blockly
npm ERR!   blockly@"7.20211209.0-beta.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer blockly@">=3.20200123.0 <4" from @blockly/plugin-workspace-search@1.20200312.2
npm ERR! node_modules/@blockly/plugin-workspace-search
npm ERR!   @blockly/plugin-workspace-search@"1.20200312.2" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /usr/local/google/home/kozbial/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /usr/local/google/home/kozbial/.npm/_logs/2021-10-01T23_22_51_519Z-debug.log
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm ERR! process terminated
npm ERR! signal SIGTERM
```
</details>

<!-- A clear and concise description of what the bug is. -->

**To Reproduce**

<!-- Explain what someone needs to do in order to see what's described above -->
Steps to reproduce the behavior:
1. `npm run test:ghpage:beta`
2. See error


**Additional context**

PR https://github.com/google/blockly-samples/pull/898 was made to update the upper bound of the blockly dependency in each of the plugins, but it fails with a similar error.

--- a/gulpfile.js
+++ b/gulpfile.js
@@ -28,8 +28,8 @@ const resolveApp = (relativePath) => path.resolve(appDirectory, relativePath);
  */
 function checkLicenses() {
   const checker = new jsgl.LicenseChecker({
-      // dev: true,
-      // verbose: false,
+    // dev: true,
+    // verbose: false,
   });
   checker.setDefaultHandlers();
   const pluginsDir = 'plugins';
@@ -162,7 +162,7 @@ function preparePlugin(pluginDir) {
           {
             label: 'README',
             link: 'README',
-          }
+          },
         ],
       })))
       .pipe(gulp.src(
@@ -219,9 +219,9 @@ function prepareExample(baseDir, exampleDir, done) {
   const assets = blocklyDemoConfig.files.filter((f) => !pageRegex.test(f));
 
   let stream = gulp.src(
-                       pages.map((f) => path.join(baseDir, exampleDir, f)),
-                       {base: baseDir, allowEmpty: true})
-                   .pipe(gulp.header(buildFrontMatter(blocklyDemoConfig)));
+      pages.map((f) => path.join(baseDir, exampleDir, f)),
+      {base: baseDir, allowEmpty: true})
+      .pipe(gulp.header(buildFrontMatter(blocklyDemoConfig)));
   if (assets.length) {
     stream = stream.pipe(gulp.src(
         assets.map((f) => path.join(baseDir, exampleDir, f)),
@@ -278,7 +278,7 @@ function deployToGhPages(repo) {
 function preparePluginsForLocal(isBeta) {
   return (done) => {
     if (isBeta) {
-      execSync(`lerna exec -- npm install blockly@beta`, {stdio: 'inherit'});
+      execSync(`lerna add blockly@beta --dev`, {stdio: 'inherit'});
     }
     execSync(`npm run boot`, {stdio: 'inherit'});
     // Bundles all the plugins.
@@ -298,7 +298,7 @@ function prepareExamplesForLocal(isBeta) {
     const examplesDirectory = 'examples';
     if (isBeta) {
       execSync(
-          `lerna exec -- npm install blockly@beta`,
+          `lerna add blockly@beta`,
           {cwd: examplesDirectory, stdio: 'inherit'});
     }
     execSync(`npm run boot`, {cwd: examplesDirectory, stdio: 'inherit'});

